services/QuillLMS/spec/models/user_spec.rb

```diff
@@ -1511,4 +1511,63 @@
       it { expect(user.reload.units_with_same_name(name)).to eq [unit] }
     end
   end
+
+  describe '#admin_sub_role' do
+    let(:user) { create(:user) }
+
+    context 'user has admin info' do
+      let!(:admin_info) { create(:admin_info, user: user)}
+
+      it 'returns the sub role from the admin info' do
+        expect(user.admin_sub_role).to eq(admin_info.sub_role)
+      end
+    end
+
+    context 'user has no admin info' do
+      it 'returns nil' do
+        expect(user.admin_sub_role).to eq(nil)
+      end
+    end
+  end
+
+  describe '#admin_approval_status' do
+    let(:user) { create(:user) }
+
+    context 'user has admin info' do
+      let!(:admin_info) { create(:admin_info, user: user)}
+
+      it 'returns the sub role from the admin info' do
+        expect(user.admin_approval_status).to eq(admin_info.approval_status)
+      end
+    end
+
+    context 'user has no admin info' do
+      it 'returns nil' do
+        expect(user.admin_approval_status).to eq(nil)
+      end
+    end
+  end
+
+  describe '#admin_sub_role=' do
+    let(:user) { create(:user) }
+
+    context 'user has admin info' do
+      let!(:admin_info) { create(:admin_info, user: user)}
+
+      it 'sets the admin info to have the new sub role' do
+        user.admin_sub_role=AdminInfo::LIBRARIAN_SLASH_MEDIA_SPECIALIST
+        expect(user.admin_sub_role).to eq(AdminInfo::LIBRARIAN_SLASH_MEDIA_SPECIALIST)
+      end
+    end
+
+    context 'user has no admin info' do
+      let!(:admin_info) { create(:admin_info, user: user)}
```

Comment:
I think we need to remove this line in order for this test to be a case when "user has no admin info"?